### PR TITLE
Add sample to analytics for UI element becoming visible

### PR DESCRIPTION
**Pull request by kul3r4**
This sample shows how to track in amp-analytics an element become visible on the page

### CODE REVIEW

File: src/20_Components/amp-analytics.html [hunk 1]
```diff
@@ -306,5 +309,40 @@
...
   </script>
   </amp-analytics>
+
+    <!-- ## UI elements appearance tracking -->
+      <!--
+    It is possible to track the appearance of an UI element. Here we have an image which is by default hidden and gets displayed 
+    after an amp-bind event. For simplicity, we are using a button. 
+      -->
+      <div>
+        <button class="ampstart-btn caps m1" on="tap:AMP.setState({hideImageTracking: false})">
```
Comment (sebastianbenz) [inline src/20_Components/amp-analytics.html]: suggestion: use the toggle action `tap:an-image.toggle`. In that case also rename the button to `toggle image`.

Comment (kul3r4) [reply src/20_Components/amp-analytics.html]: thanks i updated the sample, however I see the analytics endpoint is called twice when I click on the button

Comment (kul3r4) [reply src/20_Components/amp-analytics.html]: Ignore my previous comment...I copied and paste the same code in 2 places, this was why!

File: src/20_Components/amp-analytics.html [hunk 2]
```diff
@@ -124,6 +127,38 @@
...
+        "requests": {
+          "elementview": "https://amp-publisher-samples-staging.herokuapp.com/amp-analytics/ping?user=[[.]]&account=ampbyexample&event=${eventId}"
+        },
+        "triggers": {
+          "defaultPageview": {
+            "on": "visible",
+            "request": "elementview",
+            "selector": "#an-image",
+            "vars": {
+              "eventId": "element-ui"
```
Comment (sebastianbenz) [inline src/20_Components/amp-analytics.html]: Can you specify a more meaningful event id, one that makes it clear in the dashboard which event occured,

File: src/20_Components/amp-analytics.html [hunk 3]
```diff
@@ -124,6 +127,38 @@
...
+  <amp-analytics>
+      <script type="application/json">
+      {
+        "requests": {
+          "elementview": "https://amp-publisher-samples-staging.herokuapp.com/amp-analytics/ping?user=[[.]]&account=ampbyexample&event=${eventId}"
+        },
+        "triggers": {
+          "defaultPageview": {
+            "on": "visible",
+            "request": "elementview",
```
Comment (sebastianbenz) [inline src/20_Components/amp-analytics.html]: suggestion: s/elementview/elementShown

**Comment by kul3r4** [comment]
@sebastianbenz address the comments

File: src/20_Components/amp-analytics.html [hunk 4]
```diff
@@ -138,20 +138,20 @@
...
         },
         "triggers": {
           "defaultPageview": {
             "on": "visible",
-            "request": "elementview",
+            "request": "elementShown",
             "selector": "#an-image",
             "vars": {
-              "eventId": "element-ui"
+              "eventId": "imageview"
```
Comment (sebastianbenz) [inline src/20_Components/amp-analytics.html]: This `imageview` is very generic which makes it hard to associate the different events. If you look at the other samples, I'm using concrete events (e.g. `catview`). Maybe: `imageShownAfterButtonClick`.

**Comment by kul3r4** [comment]
@sebastianbenz PTAL

**Pull request merged**